apps/bare-expo/android/app/build.gradle

```
@@ -75,6 +75,9 @@ import com.android.build.OutputFile
  */
 
 project.ext.react = [
+    // This value can be overwritten by using the ENTRY_FILE environment variable.
+    // The Node.js script adds support for platform extensions, monorepos, and respects the package.json main field.
+    entryFile: (["node", "--print", "require('expo/scripts/resolveAppEntry')(require('path').resolve('$projectDir', '../../'), {platform:'android',relative:true})"].execute(null, rootDir).text.trim()),
```

Instead of generating JS here it seems clearer and more robust to pass in the inputs as options that the script reads from process.argv. (For one, what if projectDir contains single quotes in it?) Something like:
```suggestion
    entryFile: (["node", "expo/scripts/resolveAppEntry", $projectDir, "android"].execute(null, rootDir).text.trim()),
```

---

Don't know if node can be used like that, we'd probably need to use npx in that case 🤔

---

Node definitely can be used like this, that's the main way to use Node.

---

To address the node_modules resolution issue, this is what I propose:

```suggestion
    entryFile: (["node", "--eval", "require('expo/scripts/resolveAppEntry')(process.argv)", $projectDir, "android"].execute(null, rootDir).text.trim()),
```

1. This maximally pushes JS into .js files. Nearly all the logic is inside the script.
2. `$projectDir` is properly escaped (data is separate from commands)
3. Using `--eval` instead of `--print` treats the script as a regular script that writes to stdout with console.log. 

Using `--print` with the JSON files makes more sense to me since they aren't scripts.

---

After we adjust for relative path, this is essentially the same line. Only difference is everything else in the file works by generating JS but this will use a bin-type format.

---

I'm requesting the change for the reasons above, it closes off a latent escaping bug and moves JS into .js. Like you say it's just one line and all else equal the the robust approach is better IMO